tests/unit/test_http.py

```diff
@@ -71,6 +71,30 @@ def test_basesession__raise_for_status(mock_response):
         http.BaseSession._raise_for_status(
             mock_response(HTTPStatus.METHOD_NOT_ALLOWED, json={}))
 
+import asyncio
+import time
+@pytest.mark.asyncio
+async def test__Limiter_rate_limit():
```

Comment:
@sgillies any ideas for how we can test the rate/worker limiting? Its a tricky async/timing problem

Reply:
You mean testing the cadence? How about wrapping the limiter around calls to time.monotonic, collecting N number of times, and then asserting that the N calls don't exceed the Limiters rate limit and that no calls are closer together than the cadence threshold? The limiter can throttle anything, not only HTTP requests, right? 

```python
async def get_time_limited(limiter):
    async with limiter:
        return time.monotonic()

limiter = Limiter(...)
times = [get_time_limited(limiter) async for i in range(100)]
assert all(second - first > threshold for first, second in itertools.pairwise(times))
```

Is that useful?

Reply:
yeah, it can limit anything! This is a cool approach and is helpful.